File: go/power-of-two.go
package leetcode

func isPowerOfTwo(n int) bool {
	if n <= 0 {
		return false
	}

	found := false
	for n > 0 {
		if n&1 == 1 {
			if found {
				return false
			} else {
				found = true
			}
		}
		n >>= 1
	}
	return true
}
func isPowerOfTwo2(n int) bool {
	return n > 0 && ((n-1)&n) == 0
}

File: python/power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (n & (n-1) == 0)
    def isPowerOfTwo2(self, n: int) -> bool:
        return n > 0 and bin(n).count("1") == 1